p2p/rlpx.go

```diff
@@ -242,106 +266,77 @@ func (h *encHandshake) ecdhShared(prv *ecdsa.PrivateKey) ([]byte, error) {
 // it should be called on the dialing side of the connection.
 //
 // prv is the local client's private key.
-// token is the token from a previous session with this node.
 func initiatorEncHandshake(conn io.ReadWriter, prv *ecdsa.PrivateKey, remoteID discover.NodeID, token []byte) (s secrets, err error) {
-	h, err := newInitiatorHandshake(remoteID)
+	h := &encHandshake{initiator: true, remoteID: remoteID}
+	authMsg, err := h.makeAuthMsg(prv, token)
 	if err != nil {
 		return s, err
 	}
-	auth, err := h.authMsg(prv, token)
+	var authPacket []byte
+	if os.Getenv("RLPX_EIP8") != "" {
```

Comment:
Hmmm, or is this here so we push out the change to first only accept EIP-8, but don't initiate it and when people upgraded then switch over to use it by everyone?